File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int expectedXOR=0;
        int actualXOR=nums[0];
        int n=nums.size();
        for(int i=1;i<=n;++i) expectedXOR^=i;
        for(int i=1;i<n;++i) actualXOR^=nums[i];

        return expectedXOR^actualXOR;
    }
};

File: 268-missing-number/missing-number.py
class Solution(object):
    def missingNumber(self, nums):
        n = len(nums)

        for i in range(n):
            while nums[i] < n and nums[i] != nums[nums[i]]:
                j = nums[i] 
                nums[i], nums[j] = nums[j], nums[i]

        '''
            In Python, when you write a tuple assignment like nums[i], nums[nums[i]] = nums[nums[i]], nums[i], Python evaluates the entire right-hand side first before performing any assignments.
            This means that even if the current value of nums[i] should be skipped (e.g., nums[i] >= n), Python will still try to compute nums[nums[i]] and can raise an IndexError.
            To avoid this, you need to store nums[i] in a temporary variable first and use that in the assignment, ensuring no invalid indexing occurs.
            '''

        for i in range(n):
            if nums[i] != i:
                return i

        return n
